airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/avro_parser.py

```diff
@@ -3,13 +3,35 @@
 #
 
 import logging
-from typing import Any, Dict, Iterable
+from typing import Any, Dict, Iterable, Mapping
 
+import fastavro
 from airbyte_cdk.sources.file_based.config.file_based_stream_config import FileBasedStreamConfig
 from airbyte_cdk.sources.file_based.file_based_stream_reader import AbstractFileBasedStreamReader
 from airbyte_cdk.sources.file_based.file_types.file_type_parser import FileTypeParser
 from airbyte_cdk.sources.file_based.remote_file import RemoteFile
 
+AVRO_TYPE_TO_JSON_TYPE = {
+    "null": "null",
+    "boolean": "boolean",
+    "int": "integer",
+    "long": "integer",
+    "float": "number",
+    "double": "string",  # double -> number conversions can lose precision
+    "bytes": "string",
+    "string": "string",
+}
+
+AVRO_LOGICAL_TYPE_TO_JSON = {
+    "decimal": {"type": "string"},
+    "date": {"type": "string", "format": "date"},
+    "time-millis": {"type": "integer"},
+    "time-micros": {"type": "integer"},
+    "timestamp-millis": {"type": "string", "format": "date-time"},
+    "timestamp-micros": {"type": "string"},
+    # fastavro does not support duration https://fastavro.readthedocs.io/en/latest/logical_types.html
```

Comment:
adding!

Reply:
also noting, I spent a decent amount of time trying to figure out why datetime objects don't work with our avro io writer since tests complaining about the type not matching on file writes. For now I'll punt on this because it seems uncommon, logical types werent supported in the legacy s3, and this was only affecting our ability to test

Reply:
Thanks @brianjlai. I'm tempted to hold off on supporting logical types and throw an exception if we encounter one, until we're ready to dedicate more time to it. WDYT?

Reply:
On all logical types or just these two? I feel like most of the logical types everywhere else work pretty well and i've been able to test them with scenarios so I'm actually in favor leaving them in.